src/Features/CSharp/Portable/InitializeParameter/CSharpInitializeMemberFromParameterCodeRefactoringProvider.cs

```diff
@@ -2,45 +2,29 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Collections.Generic;
 using System.Composition;
 using System.Diagnostics.CodeAnalysis;
 using System.Threading;
 using Microsoft.CodeAnalysis.CodeRefactorings;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
-using Microsoft.CodeAnalysis.Editing;
 using Microsoft.CodeAnalysis.InitializeParameter;
-using Microsoft.CodeAnalysis.Operations;
 
 namespace Microsoft.CodeAnalysis.CSharp.InitializeParameter;
-
-using static InitializeParameterHelpersCore;
-
 [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.InitializeMemberFromParameter), Shared]
 [ExtensionOrder(Before = nameof(CSharpAddParameterCheckCodeRefactoringProvider))]
 [ExtensionOrder(Before = PredefinedCodeRefactoringProviderNames.Wrapping)]
-internal sealed class CSharpInitializeMemberFromParameterCodeRefactoringProvider :
+[method: ImportingConstructor]
+[method: SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+internal sealed class CSharpInitializeMemberFromParameterCodeRefactoringProvider() :
     AbstractInitializeMemberFromParameterCodeRefactoringProvider<
         BaseTypeDeclarationSyntax,
         ParameterSyntax,
         StatementSyntax,
         ExpressionSyntax>
 {
-    [ImportingConstructor]
-    [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
-    public CSharpInitializeMemberFromParameterCodeRefactoringProvider()
-    {
-    }
-
     protected override bool IsFunctionDeclaration(SyntaxNode node)
         => InitializeParameterHelpers.IsFunctionDeclaration(node);
 
-    protected override SyntaxNode? TryGetLastStatement(IBlockOperation? blockStatement)
-        => InitializeParameterHelpers.TryGetLastStatement(blockStatement);
-
-    protected override void InsertStatement(SyntaxEditor editor, SyntaxNode functionDeclaration, bool returnsVoid, SyntaxNode? statementToAddAfter, StatementSyntax statement)
```

Comment:
all these methods moved out of the refactoring provider into the lang service instead.  that way other refactorings providers can access them.